### PR TITLE
Add populateEntryRule in plugin's settings

### PR DESCRIPTION
**Pull request by bidoubiwa**
Fixes: https://github.com/meilisearch/strapi-plugin-meilisearch/issues/423
Fixes: https://github.com/meilisearch/strapi-plugin-meilisearch/issues/425

By default when fetching an entry from the Strapi database it will not always fetch every nested relationship. 
Imagine I have a collection that has a relationship with another collection, that itself has another relationship etc.. 
when fetching with `findOne` for example it will not go as deep. 

this PR introduces a setting that lets the user give the specific rule to provide to the fetching method to ensure that every nested relation is fetched how the user would like it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -350,6 +350,60 @@ module.exports = {
...
+
+Content-types in Strapi may have relationships with other content-types. To ensure that these links are fetched and added to an entry correctly from your Strapi database, the correct populate rule must be provided ([see documentation](https://docs-next.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/entity-service/populate.html#basic-populating)).
+
+To communicate the populate rule, use the `populate` setting on the according content-type in the plugin's settings.
+
+**For example**
+
+Imagine my `restaurant` content-type has a relation with a repeatable-component `repeatableComponent` that itself has a relationship with the content-type `categories`.
+
+The following population will ensure that a `restaurant` entry contains even the most nested.
```
Comment (bidoubiwa) [inline README.md:363]:
```suggestion
The following population will ensure that a `restaurant` entry contains even the most nested relation.
```

File: README.md [hunk 2]
```diff
@@ -350,6 +350,60 @@ module.exports = {
 
 [See resources](./resources/meilisearch-settings) for more settings examples.
 
+#### 👥 Populate entry rule
+
+Content-types in Strapi may have relationships with other content-types. To ensure that these links are fetched and added to an entry correctly from your Strapi database, the correct populate rule must be provided ([see documentation](https://docs-next.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/entity-service/populate.html#basic-populating)).
```
Comment (alallema) [inline README.md:355]: Did you mean that they have relationships with each other or do they have relationships with content-type outside of Strapi? It's not really clear

File: README.md [hunk 3]
```diff
@@ -350,6 +350,60 @@ module.exports = {
...
+    config: {
+      restaurant: {
+        populateEntryRule: ['repeatableComponent.categories', 'categories'],
+      }
+    }
+  },
+}
+```
+
+by provinding this, the following is indexed in Meilisearch:
```
Comment (alallema) [inline README.md:377]:
```suggestion
by providing this, the following is indexed in Meilisearch:
```

File: README.md (hunk 2 above)
Comment (bidoubiwa) [reply README.md]: Is it better now?

Comment (alallema) [reply README.md]: Yes !

**Review by alallema — APPROVED** [review]
LGTM! 🌯

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**